pkg/odo/util/completion/completionhandlers.go

```diff
@@ -39,3 +41,57 @@ var ServiceClassCompletionHandler = func(args complete.Args, client *occlient.Cl
 
 	return completions
 }
+
+// ProjectNameCompletionHandler provides project name completion
+var ProjectNameCompletionHandler = func(args complete.Args, client *occlient.Client, command *cobra.Command) (completions []string) {
+	completions = make([]string, 0)
+	projects, err := project.List(client)
+	if err != nil {
+		return completions
+	}
+
+	var commands []string
```

Comment:
If I return here, it would suggest the name of the projects again and again if the user keeps pressing tab again and again.
 
So I want to stop suggesting once the suggestion is complete. So I extract the flags and the commands. Check if the suggestion is already complete. If completed then return nothing else. 

Also for flags, I added a few cases.